provider/cmd/pulumi-resource-eks/schema.json

```diff
@@ -924,7 +924,8 @@
                     "description": "Type of capacity associated with the EKS Node Group. Valid values: `ON_DEMAND`, `SPOT`. This provider will only perform drift detection if a configuration value is provided."
                 },
                 "cluster": {
-                    "$ref": "#/types/eks:index:CoreData",
+                    "$ref": "#/resources/eks:index:Cluster",
+                    "plain": true,
```

Comment:
I _think_ we should be supporting both the data and resource options here ideally. 

The data version allows people to use the nodegroup components with clusters which are set up outside of EKS.

What complexities arise from making this a `oneOf`? Do you think this is feasible?

Reply:
Ah, ok. I was under the impression that the data part was more of an implementation detail that happened to "leak out" into the public API in Node, but your reasoning about why it's exposed makes sense. I'll give `oneOf` a try -- it should just work.